kafka_helpers/consumer.go

```diff
@@ -0,0 +1,50 @@
+package kafka_helpers
+
+import (
+	"context"
+	"fmt"
+	"github.com/segmentio/kafka-go"
+	"internship_project/elasticsearch_helpers"
+	"log"
+	"strings"
+)
+
+type KafkaConsumer struct {
+	Reader   *kafka.Reader
+	EsClient elasticsearch_helpers.ElasticsearchClient
+}
+
+func (consumer *KafkaConsumer) Consume() {
+	fmt.Println("KafkaConsumer is ready to consume new messages.")
+	for {
+
+		m, err := consumer.Reader.FetchMessage(context.Background())
+		if err != nil {
+			log.Printf("Error while fetching message")
```

Comment:
If a Kafka message cannot be processed (i.e. due to an error), this will lead to the whole stream getting stuck. This is obviously bad. Any ideas on how to tackle this issue?

Reply:
I think this was one addressed by Petar earlier, but here's my take on it, to check if I got it right:
- Firstly, get the message from the topic
- Send it over to ElasticSearch and make sure it's received properly and with no errors
- Commit the current offset, marking the message as consumed

Reply:
Actually there's an even bigger issue. If you get an error, message will probably be nil, causing your service to crash on m.Value access.
Even if you solve the problem above, you still have a problem if the second step of your algorithm fails. To me, it seems that if Elasticsearch operation fails, the code will go straight to committing the index. This means that the message will be skipped, and data will be lost.

Reply:
The offset is no longer commited before the message is consumed properly.

Reply:
Missing continue here.